### PR TITLE
add first version of "Falsehoods Programmers Believe About Phone Numbers"

### PR DESCRIPTION
**Pull request by dlyongemallo**

### CODE REVIEW

File: FALSEHOODS.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+1.  **Each country calling code corresponds to exactly one country**
+
+    The USA, Canada, and several Caribbean islands share the country calling code +1. Russia and Kazakhstan share +7. These are not the only examples!
+
+1.  **Each country has only one country calling code**
+
+    As of this present moment (in Mar. 2016), phones in the disputed territory and partially recognised state of Kosovo may be reached by dialing the country calling code for Serbia (+381), Slovenia (+386), or Monaco (+377), depending on where and when one obtained the number.
+
+    *Tip:* Use the phone widget to encourage users to enter their phone number in an international format such that we can understand it.
```
Comment (keghani) [inline FALSEHOODS.md]: Bold instead of italic? To be consistent with the rest (can be a separate patch)

File: FALSEHOODS.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+1.  **There are only two ways to dial a phone number: domestically and from overseas**
+
+    Some numbers may need different prefixes depending on: the carrier you are using; what device you are dialling from/to; whether you are inside or outside a particular geographical region.
+
+    Examples:
+    *   In Brazil, to dial numbers internally but across a certain geographical boundary, a carrier code must be explicitly dialed to say which carrier you will use to pay for the call.
+    *   In Nepal, the leading zero in national format is omitted depending on whether the originating phone is mobile or fixed-line.
+    *   In New Zealand, you need to dial the area-code (e.g. 03) even if the number is within the same area-code region as you are, unless it is "close" (something approximating city/district boundaries), in which case it shouldn’t be dialled.
+
+    *Tip:* Use formatForMobileDialling to get the number a user should actually dial on their mobile phone.
```
Comment (keghani) [inline FALSEHOODS.md]: Again with the *Tip:*

File: FALSEHOODS.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+1.  **To make a number dialable, you only need to change the prefix**
```
Comment (keghani) [inline FALSEHOODS.md]: I don't understand "To make a number dialable" - presumably this is going from E.164 to national format? Isn't E.164 diallable inside the country too?

File: FALSEHOODS.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+    In Argentina, to dial a mobile number domestically, the digits "15" need to be inserted *after* the area code but *before* the local number, and the "9" after the country code (54) needs to be removed. This transforms +54 9 2982 123456 into 02982 15 123456.
+
+1.  **No prefix of a valid phone number can be a valid phone number**
```
Comment (keghani) [inline FALSEHOODS.md]: I think at the negative of this, given in some places you actually can't have prefixes of numbers, there's "If I can dial a number then it's safe to store it" or similar, re. canonicalization/truncation/uniqueness.

Can be a separate patch.

File: FALSEHOODS.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
+# Falsehoods Programmers Believe About Phone Numbers
+
+... and tips on how to use libphonenumber.
+
+Given how ubiquitous phone numbers are and how long they've been around, it's surprising how many false assumptions programmers continue to make about them.
+
+1.  **Phone numbers that are valid today will always be valid. Phone numbers of a certain type today (e.g., mobile) will never be reassigned to another type.**
+
+    A phone number which connects today may be disconnected tomorrow. A number which is free to call today may cost money to call tomorrow. The phone company may decide to expand the range of available phone numbers by inserting a digit into an existing number.
```
Comment (keghani) [inline FALSEHOODS.md]: We wrapped https://raw.githubusercontent.com/googlei18n/libphonenumber/master/FAQ.md at 80 characters, should keep it consistent.

File: FALSEHOODS.md (hunk 4 above)
Comment (keghani) [inline FALSEHOODS.md]: Ah, I just saw the next one. Ignore this comment :)

File: FALSEHOODS.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+1.  **An invalid number will not reach an endpoint**
+
+    In some countries, or on some phones, extra digits are thrown away. Hence, 1-800-MICROSOFT is an invalid number - but it still connects to Microsoft, since any later digits are ignored. Numbers such as "911" can be reached by dialling "911 123" in some countries: but not in others.
+
+    In other countries, invalid numbers may be "fixed" by a carrier, e.g., adding a mobile token if they know it is a mobile number, such that it connects.
+
+1.  **All valid phone numbers follow the ITU specifications**
+
+    ITU says things like "national numbers can not be longer than sixteen digits" but valid numbers in Germany have been assigned that are longer than this.
```
Comment (keghani) [inline FALSEHOODS.md]: I think it was 15 FYI. No need to change though, you say "things *like*"

File: FALSEHOODS.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+    In Argentina, to dial a mobile number domestically, the digits "15" need to be inserted *after* the area code but *before* the local number, and the "9" after the country code (54) needs to be removed. This transforms +54 9 2982 123456 into 02982 15 123456.
+
+1.  **No prefix of a valid phone number can be a valid phone number**
+
+    In some countries, it's possible to connect to a different endpoint by dialing more digits after a number. So "12345678" may not reach the same person as dialing "123456".
+
+1.  **An invalid number will not reach an endpoint**
+
+    In some countries, or on some phones, extra digits are thrown away. Hence, 1-800-MICROSOFT is an invalid number - but it still connects to Microsoft, since any later digits are ignored. Numbers such as "911" can be reached by dialling "911 123" in some countries: but not in others.
```
Comment (keghani) [inline FALSEHOODS.md]: I'm not actually sure that vanity numbers are understood everywhere. I hadn't seem them til I lived in the States. I wonder if some readers won't understand 1-800-MICROSOFT. Maybe have a footer explaning that this would dial 1-800-648767638?

**Comment by keghani** [comment]
Looks great, thanks! I left some comments but feel free to iterate on them. You can merge.

File: FALSEHOODS.md (hunk 5 above)
Comment (dlyongemallo) [inline FALSEHOODS.md]: Sure. But the README.md and CONTRIBUTING.md are not wrapped at 80, so it's inconsistent either way.

File: FALSEHOODS.md (hunk 1 above)
Comment (dlyongemallo) [inline FALSEHOODS.md]: Done.

File: FALSEHOODS.md (hunk 2 above)
Comment (dlyongemallo) [inline FALSEHOODS.md]: Done.

File: FALSEHOODS.md (hunk 7 above)
Comment (dlyongemallo) [inline FALSEHOODS.md]: I'll have to research how vanity numbers work in non-English-speaking countries.
